hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/BaseCleanActionExecutor.java

```diff
@@ -81,9 +83,21 @@ HoodieCleanerPlan requestClean(HoodieEngineContext context) {
 
       context.setJobStatus(this.getClass().getSimpleName(), "Generates list of file slices to be cleaned");
 
-      Map<String, List<HoodieCleanFileInfo>> cleanOps = context
-          .map(partitionsToClean, partitionPathToClean -> Pair.of(partitionPathToClean, planner.getDeletePaths(partitionPathToClean)), cleanerParallelism)
-          .stream()
+      // Compute the file paths, to be cleaned in each valid file group
+      Stream<Pair<String, List<CleanFileInfo>>> cleanInfos = context.map(partitionsToClean,
+          partitionPathToClean -> Pair.of(partitionPathToClean, planner.getDeletePaths(partitionPathToClean)),
```

Comment:
is it possible to have one call to planner.getDeletePaths return all files to be cleaned? that seems like better abstraction at a high level to me. Not sure if there are disadvantages of separating them.